diff/parse.go

```diff
@@ -579,6 +593,16 @@ func linePrefix(c byte) bool {
 	return false
 }
 
+// nextLineHasPrefix peeks into the given reader to check whether the next
+// bytes match the given prefix.
+func nextLineHasPrefix(reader *bufio.Reader, prefix []byte) (bool, error) {
+	next, err := reader.Peek(len(prefix))
+	if err != nil {
```

Comment:
do we need to handle `ErrBufferFull`? I assume `len(prefix)` is always quite small.

> If Peek returns fewer than n bytes, it also returns an error explaining why the read is short

It seems like you need to handle that case. I assume by returning `false, nil`.  A test case for that would be great :)

Reply:
I just added a commit that checks for `io.EOF`, but I'm honestly not sure how to produce a test case that could make the code run into `ErrBufferFull`.